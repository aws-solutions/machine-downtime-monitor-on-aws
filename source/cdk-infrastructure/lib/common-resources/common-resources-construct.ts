// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import { Construct, RemovalPolicy } from '@aws-cdk/core';
import { BlockPublicAccess, Bucket, BucketAccessControl, BucketEncryption, CfnBucket } from '@aws-cdk/aws-s3';

export class CommonResources extends Construct {
  // S3 logging bucket
  public readonly s3LoggingBucket: Bucket;

  constructor(scope: Construct, id: string) {
    super(scope, id);

    this.s3LoggingBucket = new Bucket(this, 'LogBucket', {
      accessControl: BucketAccessControl.LOG_DELIVERY_WRITE,
      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
      encryption: BucketEncryption.S3_MANAGED,
      removalPolicy: RemovalPolicy.RETAIN
    });
    const cfnBucket = this.s3LoggingBucket.node.defaultChild as CfnBucket;
    cfnBucket.addMetadata('cfn_nag', {
      rules_to_suppress: [
        { id: 'W35', reason: 'This bucket is to store S3 logs, so it does not require access logs.' },
        { id: 'W51', reason: 'This bucket is to store S3 logs, so it does not require S3 policy.' }
      ]
    });
  }
}