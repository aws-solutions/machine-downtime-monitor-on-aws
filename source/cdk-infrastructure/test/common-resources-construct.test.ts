// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import '@aws-cdk/assert/jest';
import { SynthUtils } from '@aws-cdk/assert';
import { Stack } from '@aws-cdk/core';
import { CommonResources } from '../lib/common-resources/common-resources-construct';

test('Common resources snapshot and variables', () => {
  // PREPARE
  const stack = new Stack();

  // WHEN
  const commonResources = new CommonResources(stack, 'TestCommonResources');

  // THEN
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  expect(commonResources.s3LoggingBucket).toBeDefined();
  expect(stack).toHaveResourceLike('AWS::S3::Bucket', {
    AccessControl: 'LogDeliveryWrite',
    BucketEncryption: {
      ServerSideEncryptionConfiguration: [{
        ServerSideEncryptionByDefault: { SSEAlgorithm: 'AES256' }
      }]
    },
    PublicAccessBlockConfiguration: {
      BlockPublicAcls: true,
      BlockPublicPolicy: true,
      IgnorePublicAcls: true,
      RestrictPublicBuckets: true
    }
  });
});