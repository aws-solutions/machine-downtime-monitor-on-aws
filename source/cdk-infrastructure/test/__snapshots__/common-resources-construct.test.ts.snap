// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Common resources snapshot and variables 1`] = `
Object {
  "Resources": Object {
    "TestCommonResourcesLogBucket77572DDE": Object {
      "DeletionPolicy": "Retain",
      "Metadata": Object {
        "cfn_nag": Object {
          "rules_to_suppress": Array [
            Object {
              "id": "W35",
              "reason": "This bucket is to store S3 logs, so it does not require access logs.",
            },
            Object {
              "id": "W51",
              "reason": "This bucket is to store S3 logs, so it does not require S3 policy.",
            },
          ],
        },
      },
      "Properties": Object {
        "AccessControl": "LogDeliveryWrite",
        "BucketEncryption": Object {
          "ServerSideEncryptionConfiguration": Array [
            Object {
              "ServerSideEncryptionByDefault": Object {
                "SSEAlgorithm": "AES256",
              },
            },
          ],
        },
        "PublicAccessBlockConfiguration": Object {
          "BlockPublicAcls": true,
          "BlockPublicPolicy": true,
          "IgnorePublicAcls": true,
          "RestrictPublicBuckets": true,
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
